lua/neoscroll/init.lua

```diff
@@ -35,7 +35,9 @@ end
 local function hide_cursor()
     if vim.o.termguicolors then
         guicursor = vim.o.guicursor
-        vim.o.guicursor = guicursor .. ',a:NeoscrollHiddenCursor'
+	if guicursor ~= '' then
+            vim.o.guicursor = guicursor .. ',a:NeoscrollHiddenCursor'
```

Comment:
Now that we are at it, I think hiding the cursor should just work by setting `guicursor` to `'a:NeoscrollHiddenCursor'` directly (no need to append it to the current `guicursor`). So change this line to `vim.o.guicursor = 'a:NeoscrollHiddenCursor'`